File: MiniDOMBase/DMD/Anagrafica/Distributori/Distributori.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases
Imports minidom.Anagrafica
Imports minidom.Internals

Namespace Internals


    Public NotInheritable Class CDistributoriClass
        Inherits CModulesClass(Of CDistributore)

        Friend Sub New()
            MyBase.New("modDistributori", GetType(CDistributoriCursor), -1)
        End Sub


        Public Function GetItemByName(ByVal value As String) As CDistributore
            value = Trim(value)
            If (value = vbNullString) Then Return Nothing
            For Each item As CDistributore In Me.LoadAll
                If Strings.Compare(item.Nome, value) = 0 Then Return item
            Next
            Return Nothing
        End Function

    End Class

End Namespace

Partial Public Class Anagrafica



    Private Shared m_Distributori As CDistributoriClass = Nothing

    Public Shared ReadOnly Property Distributori As CDistributoriClass
        Get
            If (m_Distributori Is Nothing) Then m_Distributori = New CDistributoriClass
            Return m_Distributori
        End Get
    End Property

End Class

File: MiniDOMBase/DMD/Anagrafica/Distributori/Distributori.cs
using System;
using DMD;
using DMD.Databases;
using static minidom.Sistema;
using static minidom.Anagrafica;
using minidom.repositories;
using DMD.Databases.Collections;


namespace minidom
{
    namespace repositories
    {

        /// <summary>
        /// Repository di oggetti <see cref="CDistributore"/>
        /// </summary>
        [Serializable]
        public sealed class CDistributoriClass 
            : CModulesClass<Anagrafica.CDistributore>
        {
            
            /// <summary>
            /// Costruttore
            /// </summary>
            public CDistributoriClass()
                : base("modDistributori", typeof(Anagrafica.CDistributoriCursor), -1)
            {
            }

            /// <summary>
            /// Restituisce l'elemento in base al nome
            /// </summary>
            /// <param name="value"></param>
            /// <returns></returns>
            public Anagrafica.CDistributore GetItemByName(string value)
            {
                value = DMD.Strings.Trim(value);
                if (string.IsNullOrEmpty(value))
                    return null;
                foreach (Anagrafica.CDistributore item in LoadAll())
                {
                    if (DMD.Strings.Compare(item.Nome, value, true) == 0)
                        return item;
                }

                return null;
            }
        }
    }

    public partial class Anagrafica
    {
        private static CDistributoriClass m_Distributori = null;

        /// <summary>
        /// Repository di oggetti <see cref="CDistributore"/>
        /// </summary>
        public static CDistributoriClass Distributori
        {
            get
            {
                if (m_Distributori is null)
                    m_Distributori = new CDistributoriClass();
                return m_Distributori;
            }
        }
    }
}